### PR TITLE
api: add grpc and diagnostics

### PR DESCRIPTION
**Pull request by xhebox**
Issue Number: ref #387

Problem Summary: GRPC service is implemented by `github.com/pingcap/sysutil`. So this PR is trvial. Main changes are in `pkg/server/api/http.go`.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Notable changes

- [ ] Has configuration change
- [ ] Has HTTP API interfaces change
- [ ] Has tiproxyctl change
- [ ] Other user behavior changes

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Support diagnosticspb grpc service for cluster_xxx system tables
```

### CODE REVIEW

File: pkg/server/api/http.go [hunk 1]
```diff
@@ -42,27 +49,43 @@ type managers struct {
 	crt *mgrcrt.CertManager
 }
 
-type HTTPServer struct {
+type Server struct {
```
Comment (djshow832) [inline pkg/server/api/http.go:52]: Maybe the filename should also be renamed?
Or move the grpc part to another file?

**Review by djshow832 — COMMENTED** [review]
Can you add some tests? The test coverage of this pkg is still 0.

File: pkg/server/api/http.go (hunk 1 above)
Comment (xhebox) [reply pkg/server/api/http.go:51]: Renamed.

**Comment by djshow832** [comment]
Fix CI

**Comment by xhebox** [comment]
> Fix CI

Done

**Review by djshow832 — APPROVED** [review]

**Pull request merged**